src/protocol/attribution/mod.rs

```diff
@@ -40,6 +41,52 @@ pub struct AggregateCreditOutputRow<F: Field> {
     credit: Replicated<F>,
 }
 
+impl<F: Field> AggregateCreditOutputRow<F> {
+    // TODO: this gets repetitive. For the most of our structures we need to be able to serialize
```

Comment:
yeah this kinda feels like you want to recreate the `serde::Serialize/Deserialize` macros...